src/client/datascience/interactive-window/interactiveWindow.ts

```diff
@@ -150,55 +147,175 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
         this.cellMatcher = new CellMatcher(this.configuration.getSettings(this.owningResource));
     }
 
-    private async createKernelReadyPromise(): Promise<IKernel> {
-        const editor = await this._editorReadyPromise;
-        const controller = await this._controllerReadyPromise.promise;
-        initializeInteractiveOrNotebookTelemetryBasedOnUserAction(this.owner, controller.connection);
-        const kernel = this.kernelProvider.getOrCreate(editor.document, {
-            metadata: controller.connection,
-            controller: controller.controller,
-            resourceUri: this.owner
-        });
-        kernel.onRestarted(
-            async () => {
-                traceInfoIfCI('Restart event handled in IW');
+    private async createReadyPromise(
+        originalConnection: KernelConnectionMetadata | undefined
+    ): Promise<NotebookEditor> {
+        // Find our preferred controller
+        const preferredController = await this.getPreferredController(originalConnection);
+
+        // Create our editor using that controller
+        const editor = await this.createEditor(preferredController);
+
+        // If that worked, insert our first sys info message
+        if (editor && preferredController) {
+            await this.insertSysInfoMessage(editor.document, preferredController.connection, SysInfoReason.Start);
+
+            // Also start connecting to our kernel but don't wait for it to finish
+            this.startKernel(editor.document, preferredController).ignoreErrors();
+        }
+
+        return editor;
+    }
+
+    private async startKernel(notebook: NotebookDocument, controller: VSCodeNotebookController): Promise<void> {
+        if (controller.id !== this._kernelConnectionId) {
+            this._kernelConnectionId = controller.id;
```

Comment:
When will this be false? Is it when we're already connected to the kernel? Will there be a "connecting" sys info left behind in that case, and should we ensure that the _kernelPromise is resolved?

Reply:
This is false on the first startup and if the kernel fails to have ipykernel and the user decides to switch to a new controller. Then it will reenter here and try again.